File: exercism/javascript/complete/pascals-triangle/pascals-triangle.js
const Triangle = (level) => {
  let rows = [[1]]

  for (let i = 1; i < level; i++) {
    let array = []
    for (let x = 0; x <= rows[i-1].length; x++) {
      if (x == 0 || x == rows[i-1].length) {
        array.push(1)
      } else {
        array.push(rows[i-1][x-1] + rows[i-1][x])
      }
    }
    rows.push(array)
  }
  const lastRow = rows[rows.length-1]
  return { rows, lastRow }
}

export default Triangle

File: exercism/ruby/complete/pascals-triangle/pascals_triangle.rb
class Triangle
  attr_reader :rows

  def initialize(level)
    @rows = [[1]]
    for i in 1...level
      row = []
      n = @rows[i-1].length
      for j in 0..n
        if j == n
          row << @rows[i-1][j-1]
        elsif j == 0
          row << @rows[i-1][j]
        else
          row << @rows[i-1][j-1] + @rows[i-1][j]
        end
      end
      @rows << row
    end
  end
end
